src/CoreWCF.Primitives/src/CoreWCF/Security/SecurityProtocolFactory.cs

```diff
@@ -0,0 +1,1085 @@
+            this.communicationObject = new WrapperSecurityCommunicationObject(this);
+        }
+
+        internal SecurityProtocolFactory(SecurityProtocolFactory factory)
+            : this()
+        {
+            if (factory == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("factory");
+            }
+
+            this.actAsInitiator = factory.actAsInitiator;
+            this.addTimestamp = factory.addTimestamp;
+            this.detectReplays = factory.detectReplays;
+            this.incomingAlgorithmSuite = factory.incomingAlgorithmSuite;
+            this.maxCachedNonces = factory.maxCachedNonces;
+            this.maxClockSkew = factory.maxClockSkew;
+            this.outgoingAlgorithmSuite = factory.outgoingAlgorithmSuite;
+            this.replayWindow = factory.replayWindow;
+            this.channelSupportingTokenAuthenticatorSpecification = new Collection<SupportingTokenAuthenticatorSpecification>(new List<SupportingTokenAuthenticatorSpecification>(factory.channelSupportingTokenAuthenticatorSpecification));
+            this.scopedSupportingTokenAuthenticatorSpecification = new Dictionary<string, ICollection<SupportingTokenAuthenticatorSpecification>>(factory.scopedSupportingTokenAuthenticatorSpecification);
+            this.standardsManager = factory.standardsManager;
+            this.timestampValidityDuration = factory.timestampValidityDuration;
+           // this.auditLogLocation = factory.auditLogLocation;
+            this.suppressAuditFailure = factory.suppressAuditFailure;
+           // this.serviceAuthorizationAuditLevel = factory.serviceAuthorizationAuditLevel;
+           // this.messageAuthenticationAuditLevel = factory.messageAuthenticationAuditLevel;
+            if (factory.securityBindingElement != null)
+            {
+                this.securityBindingElement = (SecurityBindingElement) factory.securityBindingElement.Clone();
+            }
+            this.securityTokenManager = factory.securityTokenManager;
+            this.privacyNoticeUri = factory.privacyNoticeUri;
+            this.privacyNoticeVersion = factory.privacyNoticeVersion;
+            this.endpointFilterTable = factory.endpointFilterTable;
+            this.extendedProtectionPolicy = factory.extendedProtectionPolicy;
+            this.nonceCache = factory.nonceCache;
+        }
+
+        internal WrapperSecurityCommunicationObject CommunicationObject => this.communicationObject;
+
+        // The ActAsInitiator value is set automatically on Open and
+        // remains unchanged thereafter.  ActAsInitiator is true for
+        // the initiator of the message exchange, such as the sender
+        // of a datagram, sender of a request and sender of either leg
+        // of a duplex exchange.
+        public bool ActAsInitiator => this.actAsInitiator;
```

Comment:
Can you remove the ActAsInitiator and cleanup everywhere that references it. If it looks like that would be a lot of changes, open an issue to clean up the ActAsInitiator code

Reply:
agree..https://github.com/CoreWCF/CoreWCF/issues/258